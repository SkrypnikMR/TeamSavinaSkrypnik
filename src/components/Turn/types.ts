export type TTurn = {
    login?: string;
    turn?: string;
};
