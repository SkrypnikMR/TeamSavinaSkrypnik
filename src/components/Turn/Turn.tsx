import React from 'react';
import { useTranslation } from 'react-i18next';
import { TTurn } from './types';

const Turn = ({ turn, login }: TTurn) => {
    const { t } = useTranslation();
    return turn === login
        ? <p>{t('your_turn')}</p>
        : <p>{t('not_your_turn')}</p>;
};

export default Turn;
