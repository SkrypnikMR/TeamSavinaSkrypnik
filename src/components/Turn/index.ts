import { connect } from 'react-redux';
import Turn from './Turn';
import { getStepOrderSelector, getUserLogin } from '../../store/game/selectors';

const mapStateToProps = store => ({
    turn: getStepOrderSelector(store),
    login: getUserLogin(store),
});

export default connect(mapStateToProps)(Turn);
