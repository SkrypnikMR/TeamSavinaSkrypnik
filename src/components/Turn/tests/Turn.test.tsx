import React from 'react';
import { shallow, mount } from 'enzyme';
import Turn from '../Turn';

describe('Turn', () => {
    const props = {
            turn: 'login1',
            login: 'login2',
          };
    it('Should match snapshot', () => {
        const component = shallow(<Turn />);
        expect(component.html()).toMatchSnapshot();
    });
    it('Should turn prop !== login', () => {
        const component = mount(<Turn {...props} />);
        expect(component.find('p').text()).toBe('not_your_turn');
    });
});
