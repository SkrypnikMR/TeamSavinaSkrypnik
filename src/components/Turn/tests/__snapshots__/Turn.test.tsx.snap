// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Turn Should match snapshot 1`] = `"<p>your_turn</p>"`;
